src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -1019,6 +1018,23 @@ cpdef object compute_network_structured(
                 #tuple of MC_Reach and reach_type
                 MC_Reach(segment_objects, array('l',upstream_ids))
                 )
+            
+    cdef np.ndarray fill_index_mask = np.ones_like(data_idx, dtype=bool)
+    cdef Py_ssize_t fill_index
+    cdef long upstream_tw_id
+    cdef dict tmp
+    cdef int idx
+    cdef float val
+    
+    for upstream_tw_id in upstream_results:
+        tmp = upstream_results[upstream_tw_id]
+        fill_index = tmp["position_index"]
+        fill_index_mask[fill_index] = False
+        for idx, val in enumerate(tmp["results"]):            
```

Comment:
I'm wondering if this loop would be easier to read of we wrote it to only loop over timesteps like this:
```python
#before the loop
cdef float[:] results

#inside loop
results = tmp['results']
for ts_idx in range(0, tmp['results'].shape[0], 3):
    ts_off = ts_idx * 3
    flowveldepth_nd[fill_index, ts_idx + 1, :] = results[ts_off:ts_off+3]
```
Or we can use the idea of partitioning an array into timesteps and simply iterate over blocks of values as timesteps. We could use toolz.partition, but that might introduce some overhead as it returns tuples, and we would prefer array slices.
```python
def iterate_timesteps(width, arr):
    if len(arr) % width != 0:
        raise ValueError(f"{width} doesn't evenly partition {len(arr)}")
    start = 0
    while start < len(arr):
        end = start + width
        yield arr[start:end]
        start = end
```
```python
cpdef float[:] results

for tsi, ts in enumerate(iterate_timesteps(3, tmp['results']), 1):
    flowveldepth_nd[fill_index, tsi, :] = ts
```